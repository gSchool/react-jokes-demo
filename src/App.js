import React, { Component } from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import './App.css';
import Jokes from './components/Jokes'
import { getJokes } from './actions'
import AddJokeForm from './containers/AddJokeForm'

class App extends Component {

  async componentDidMount() {
    this.props.getJokes()
  }

  render() {
    return (
      <div className="App">
        <h1>Jokes</h1>
        <Jokes />
        <AddJokeForm />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({ getJokes }, dispatch)

export default connect(null, mapDispatchToProps)(App);
