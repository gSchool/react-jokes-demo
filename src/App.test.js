import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import renderer from 'react-test-renderer'

const jokesData = [{id:1, question:"What’s a penguin’s favorite salad?", answer:"Iceberg lettuce."}]

const middlewares = [ thunk ]
const mockStore = configureMockStore(middlewares)
const store = mockStore({
  jokes: jokesData
})

describe('App', () => {
  afterEach(() => {
    fetchMock.restore()
  })

  beforeEach(() => {
    fetchMock.get('end:/jokes', jokesData
    )
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the same way every time', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
