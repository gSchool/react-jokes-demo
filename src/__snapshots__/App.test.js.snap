// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the same way every time 1`] = `
<div
  className="App"
>
  <h1>
    Jokes
  </h1>
  <div
    className="joke-list"
  >
    <h1>
      Penguin Jokes
    </h1>
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        <b>
          What’s a penguin’s favorite salad?
        </b>
        <p>
          Iceberg lettuce.
        </p>
      </li>
    </ul>
  </div>
  <form
    className="well"
    onSubmit={[Function]}
  >
    <fieldset
      className="form-group"
    >
      <label
        htmlFor="question"
      >
        Question
      </label>
      <input
        className="form-control"
        name="question"
        type="text"
      />
      <label
        htmlFor="answer"
      >
        Answer
      </label>
      <input
        className="form-control"
        name="answer"
        type="text"
      />
      <button
        className="btn btn-success"
      >
        Add
      </button>
    </fieldset>
  </form>
</div>
`;
